web/src/store/index.js

```diff
@@ -151,7 +153,7 @@ const mutations = {
    */
   [SET_DATASET_DATA](state, { data }) {
     const {
-      id, name, size, created, description,
+      id, name, size, created, description, selected_columns: selectedColumns,
```

Comment:
```suggestion
      id, name, size, created, description,
```

Reply:
I don't think there's a need to extract this at the top unless the payload key matches the dataset property key exactly.